### DISCUSSION TITLE
Transfers into liabilties show up negativ

### DISCUSSION THREAD

**Discussion by hr87** [firefly-iii/firefly-iii#6734] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Transactions from asset accounts into liabilities show up negative in both. The should be negative in the asset account and positive in the liability. 

Currently, transactions like these are a withdrawal, and handles as such on both sides. They should be transfers.

### Expected behaviour

The transactions should be transfers, negative on one side, positive on the other.

### Debug information

Happens on the demo site

### Steps to reproduce

Create a transaction in the demo side from an asset account into a liability. It shows up with the same sign on both sides.

**Comment by JC5** [comment]
Yes, they will always show up negative. That's intentional and not a bug.

**Reply by hr87** [reply]
It is negative on the asset account side, so it should be positive on the liability account side. And the graph and total balance it is counted positive, but in the transaction listings it shows up negative
![Screenshot from 2022-12-26 09-54-28](https://user-images.githubusercontent.com/12700929/209569582-6d0b0388-76e1-4a25-8594-7ffc77fcba85.png)

This screen shot shows the demo side with money being transferred from the checking account into the liability. The graph goes up, so positive 100. But the transaction shows up at negative 100. It is also counted against the spending, as you can see on the right. But the money was not spend, just transferred from one account into another.

That is why I suggest to turn transactions between assets and liabilities into transfers.

**Reply by JC5** [reply]
> It is negative on the asset account side, so it should be positive on the liability account side.

Yes, they will always show up negative. That's intentional and not a bug.

**Reply by hr87** [reply]
Why does it show up negative? And if it is intentionally, then it should also show up negative in the balance of the liability, but it is counted positive there.

Basically, if one sums up all the transactions in the account, one should get the account balance. But currently the account balance is 100, and the sum of transactions is -100. 

The other problem I see is, if money gets moved between accounts, the balance change should be 0, since the money is just booked in a different account. Currently, it is -200, the sum over both transaction entries, though.

**Reply by JC5** [reply]
https://docs.firefly-iii.org/firefly-iii/concepts/accounts/#liabilities

If you put money into a liability, it's an expense, that's why it's negative and red. However, the liability will get its balance increased. After all, that's where the money goes to.

From there, it depends on the liability and the demo site is not the best example. But if you owe money (ie a mortgage) the amount in the mortage would go up, ie. from -100.000 to -99.900. 

Since the account on the demo site is at zero, the amount is positive.

**Reply by hr87** [reply]
Moving money into a liability is not an expense. If one has a mortgage with a balance of 100 000, and a checking account with a balance of 10 000, then the total net worth is -90 000. Now if 5k are moved from checking to mortgage, the net worth is still -90 000, but now the mortgage is 95 000 and the checking account is 5 000. No money has been spend.

The expense is buying the house. So if one opens the liability, the initial balance is 0. Then the money for the house is drawn, which puts the mortgage into negative. 

You even describe that in the documentation you quote, but for the opposite case, that somebody hows you money, but it also applies to a mortgage. Move money from the liability to an expense account, the actual expense, changing the networth. Paying that off, are transfers, that leave the total networth unchanged.

I do not expect firefly to be a perfect accounting software, but I think it should at least honor the most basic rule I mentioned above: The current balance of an account is the sum of all transactions in that account (opening balance counts as transaction). And currently liabilities break this basic rule.

**Reply by JC5** [reply]
Luckily, there's a page for that as well.

https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/personal-finances/#accounting-versus-personal-finance

**Reply by hr87** [reply]
Sorry for the late reply, got distracted.

I saw this in your documentation. But I think there is a difference between not everything correct in an accounting type of sence (e.g., if buying a house is an expense, asset or equity, the sign of liabilities is flipped, credit cards are liabilities, not assets, money loaned to somebody is an asset) and inconsistencies, that break the usage of Firefly. Let me try to give an example with a mortgage account.

| Description      | Type     | Amount | Shows up as | Balance | Net worth | Running Balance | Sum Expenses |
|------------------|----------|--------|-------------|---------|----------|-----------------|--------------|
| Opening Account  |   Magic       | 0      |             | 0       | 0        | 0               |              |
| Buy House        | Withdrawal | 100000 | -100000     | -100000 | -100000  | -100000         | 100000       |
| Mortgage Payment | Withdrawal | 1000   | -1000       | -99000  | -100000  | -101000         | 101000       |
| Mortgage Payment | Withdrawal | 1000   | -1000       | -98000  | -100000  | -102000         | 102000       |

This is how it currently is. Here I only focus on the liability account. I assume the mortgage payments are done from an asset account, but I ignore that in the total net worth. I also purposefully ignored interest and other part of mortgage payments (which are normally plain expenses). There are multiple inconsistencies. 

You open the account, you buy the house, which is logged as an expense. Now every payments is also logged as an expense. This results in a wrong expense amount for the house (double in the end). Also, each mortgage payment is logged as an expense, but it does not change the total net worth. The running balance of the transaction in the account also comes out as double at the end, and is not equal to the balance of the account. Also, you have expenses without actually changing the net worth.

| Description      | Type     | Amount | Shows up as | Balance | Net worth | Running Balance | Sum Expenses |
|------------------|----------|--------|-------------|---------|----------|-----------------|--------------|
| Opening Account  |     Magic     | 100000 | -100000     | -100000 | -100000  | -100000         | 0       |
| Mortgage Payment | Withdrawal | 1000   | -1000       | -99000  | -100000  | -101000         | 1000       |
| Mortgage Payment | Withdrawal | 1000   | -1000       | -98000  | -100000  | -102000         | 2000       |

Now you can argue that buying the house is the opening balance, as seen in this table. Still not consistent. Now the final expense amount is correct, but the remaining points still stand. And you have a change in net worth without an expense from the opening transaction, you still have expenses without change in net worth and the running balance is still wrong.

| Description      | Type       | Amount | Shows up as | Balance | Running Balance | Net worth | Sum Expenses |
|------------------|------------|--------|-------------|---------|-----------------|----------|--------------|
| Opening Account  | Magic      | 0      | 0           | 0       | 0               | 0        |              |
| Buy House        | Withdrawal | 100000 | -100000     | -100000 | -100000         | -100000  | 100000       |
| Mortgage Payment | Transfer   | 1000   | 1000        | -99000  | -99000          | -100000  | 100000       |
| Mortgage Payment | Transfer   | 1000   | 1000        | -98000  | -98000          | -100000  | 100000       |

Now if we change the Mortgage payments to transfers from the asset account to the liability account, all the inconsistencies go away. The running balance is the actual balance. You only have change in net worth when there is an expense, and no change in net worth without an expense. The total expense is correct. And the nice thing with this is, if one wants to see the house as an asset, one can do that to, and it still works.

Final thing, if one thinks of the mortgage just as a monthly expense, that can easily be achieved by using an expense account for the mortgage.

**Comment by diamant-x** [comment]
Sorry to post in this old topic, but maybe just to bring in another point of view. I'm also giving thoughts about deposits and withdrawals. The main summary i'm using to look at the data is basically, "deposits are money you have economic control over" and "withdrawals are money you no longer have economic control over it". This helps me read the same way for when doing treasury bonds investment: when I buy them is an expense, because I no longer have control of it, (although i'm certainly expecting to get it back at some point), and when i get it back is a deposit (although it was previously already 'old money') because is the moment i'm recovering 'control' over it.

> You open the account, you buy the house, which is logged as an expense.

Regarding this, you miss the previous step, which is to receive the money from the mortgage in the account (A deposit from your bank). Then you use that money to buy the house (expense). Both of those should 'balance out' (not double in the end as you mention), and then you are left with the mortgage payments, which are again expenses and show up as such when done against the liability 'account'. Which would end up summing all the money spent in the mortgage. In this scenario, you could do the first deposit of the money the bank gives you come from this liability account and then you get your starting point of the account (and counting towards net worth). In my case I create an additional opening balance to account for the interests I'm expected to pay (fix rate) so that at the end I end up with a balance of zero, and meanwhile I can see the reality of what i'm pending to pay.

Another conversation, in a different topic, would be that the expense to buy a house could potentially be a transfer to an 'asset account' reflecting an investment. I would say this is a tricky way to see it, as then you would need to ideally keep asking for a quotation on your home value over time to reflect depreciation and/or increase in value. There are many other discussions about handling 'investments' in FFIII you may want to take a look at.

**Comment by grayfrost90** [comment]
I don't really mind how the payments are displayed, as long as it makes sense and adds up..... which is where my query around this comes in. I have both payments going into my Mortgage liability and interest being added to it. The Mortgage is a negative amount I need to pay off. Payments made on the debt bring it closer to 0 and interest moves it further into the negative amount - that is how my online banking shows it.

however all transactions for the liability in Firefly show as negative - both payments from an asset account and interest on the debt, yet the graphs are correct - going up (closer to 0) when I make a payment and down (further into the negative) when there's interest on the debt. It's a bit confusing when you look at the transactions 🥴

**Comment by avee87** [comment]
Same sentiment here.
Handling debt repayment as expenses leads to double-counting. If you have a dashboard showing monthly expenses it will show wildly incorrect data for mortgage - i.e., instead of `interest` it will show `monthly repayment + interest` which doesn't make sense no matter how you look at it. And the total sum of expenses for the lifetime of the mortgage will be `2 * (principal + interest)`.

I think I'll be migrating all my liability accounts to assets to have correct accounting on them...
